packages/lib-sourcify/src/lib/solidityCompiler.ts

```diff
@@ -29,40 +42,47 @@ export async function useCompiler(version: string, solcJsonInput: JsonInput) {
   // Not possible to retrieve compilers with "-ci.".
   if (version.includes('-ci.')) version = version.replace('-ci.', '-nightly.');
   const inputStringified = JSON.stringify(solcJsonInput);
-  const solcPath = await getSolcExecutable(version);
   let compiled: string | undefined;
 
-  if (solcPath) {
-    const shellOutputBuffer = spawnSync(solcPath, ['--standard-json'], {
-      input: inputStringified,
-      maxBuffer: 1000 * 1000 * 10,
-    });
-
-    // Handle errors.
-    let error: false | Error = false;
-    if (shellOutputBuffer.error) {
-      const typedError: NodeJS.ErrnoException = shellOutputBuffer.error;
-      // Handle compilation output size > stdout buffer
-      if (typedError.code === 'ENOBUFS') {
-        error = new Error('Compilation output size too large');
+  const solcPlatform = findSolcPlatform();
+  if (solcPlatform) {
+    const solcPath = await getSolcExecutable(solcPlatform, version);
+    if (solcPath) {
+      const shellOutputBuffer = spawnSync(solcPath, ['--standard-json'], {
+        input: inputStringified,
+        maxBuffer: 1000 * 1000 * 10,
+      });
+
+      // Handle errors.
+      let error: false | Error = false;
+      if (shellOutputBuffer.error) {
+        const typedError: NodeJS.ErrnoException = shellOutputBuffer.error;
+        // Handle compilation output size > stdout buffer
+        if (typedError.code === 'ENOBUFS') {
+          error = new Error('Compilation output size too large');
+        }
+        error = new Error('Compilation Error');
       }
-      error = new Error('Compilation Error');
-    }
-    if (!shellOutputBuffer.stdout) {
-      error = new Error(RECOMPILATION_ERR_MSG);
-    }
-    if (error) {
-      console.error(error);
-      throw error;
+      if (!shellOutputBuffer.stdout) {
+        error = new Error(RECOMPILATION_ERR_MSG);
+      }
+      if (error) {
+        console.error(error);
+        throw error;
+      }
+      compiled = shellOutputBuffer.stdout.toString();
     }
-    compiled = shellOutputBuffer.stdout.toString();
   } else {
     const soljson = await getSolcJs(version);
-    compiled = soljson.compile(inputStringified);
+    if (soljson) {
+      compiled = soljson.compile(inputStringified);
+    }
   }
 
   if (!compiled) {
-    throw new Error('Compilation failed. No output from the compiler.');
+    throw new Error(
+      'Compilation failed. No output from the compiler. Try using a different compiler.'
```

Comment:
```suggestion
      'Compilation failed. No output from the compiler.'
```

You can't compile a contract with a different _version_ of the compiler.

If we mean different _platform_ of the compiler, it wouldn't make sense because we're not giving the option to the user to choose. I suggest removing the last sentence

Reply:
Yeah I left it by mistake from the previous implementation